ThingIFSDK/ThingIFSDK/Predicate.swift

```diff
@@ -9,7 +9,7 @@
 import Foundation
 
 /** Protocol represents Predicate */
-public protocol Predicate :  NSCoding {
+public protocol Predicate: class, NSCoding {
```

Comment:
@yongpingchen 

I have 2 reasons:

1. If we extends `class` to protocols, it is more descriptive. We can claims the protocol is used only for class.
1. Extending `class` to protocols enables to use an instance of the class as weak reference.

I found second reason on [here](http://qiita.com/shimesaba/items/e0c3ae189ca2b247fa98).